engine/services/engine_service.cc

```diff
@@ -76,16 +76,42 @@ std::vector<EngineInfo> EngineService::GetEngineInfoList() const {
 void EngineService::InstallEngine(const std::string& engine,
                                   const std::string& version) {
   auto system_info = system_info_utils::GetSystemInfo();
+  auto get_params = [&engine, &version]() -> std::vector<std::string> {
+    if (version == "latest") {
+      return {"repos", "janhq", engine, "releases", version};
+    } else {
+      return {"repos", "janhq", engine, "releases"};
+    }
+  };
+
   auto url_obj = url_parser::Url{
       .protocol = "https",
       .host = "api.github.com",
-      .pathParams = {"repos", "janhq", engine, "releases", version},
+      .pathParams = get_params(),
   };
 
   httplib::Client cli(url_obj.GetProtocolAndHost());
   if (auto res = cli.Get(url_obj.GetPathAndQuery());
       res->status == httplib::StatusCode::OK_200) {
     auto body = json::parse(res->body);
+    auto get_data =
+        [&version](const nlohmann::json& json_data) -> nlohmann::json {
```

Comment:
Just a question. Is `std::optional<nlohmann::json>` is better? Can just return `std::nullopt` if not found instead of object `nlohmann::json()`

Reply:
We assign new value to `body` if version is not `latest`, so I think it does not make sense to use std::optional in this case.